internal/clients/kafka/acl/acl.go

```diff
@@ -0,0 +1,235 @@
+package acl
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/crossplane-contrib/provider-kafka/apis/acl/v1alpha1"
+
+	"github.com/pkg/errors"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"github.com/twmb/franz-go/pkg/kmsg"
+)
+
+// AccessControlList is a holistic representation of a Kafka ACL with configurable
+// fields
+type AccessControlList struct {
+	ResourceName              string
+	ResourceType              string `json:"ResourceType"`
+	ResourcePrincipal         string `json:"ResourcePrincipal"`
+	ResourceHost              string `json:"ResourceHost"`
+	ResourceOperation         string `json:"ResourceOperation"`
+	ResourcePermissionType    string `json:"ResourcePermissionType"`
+	ResourcePatternTypeFilter string `json:"ResourcePatternTypeFilter"`
+}
+
+// List lists all the ACLs in Kafka
+func List(ctx context.Context, cl *kadm.Client, accessControlList *AccessControlList) (*AccessControlList, error) {
```

Comment:
Linting is coming up with one more issue `cyclomatic complexity 11 of func `List` is high (> 10)`

Cyclomatic complexity is used as a metric for the complexity of tests required and we see it often in controllers. (see https://github.com/fzipp/gocyclo).

Maybe in future releases some of the logic can be factored out into functions. 

To fix, add the ` // nolint:gocyclo` comment to the function like so:

```
func List(ctx context.Context, cl *kadm.Client, accessControlList *AccessControlList) (*AccessControlList, error) { // nolint:gocyclo
```